frontend/src/component/suggestChanges/SuggestedChangeOverview/SuggestedChangeTimeline/SuggestedChangeTimeline.tsx

```diff
@@ -1,52 +1,136 @@
 import TimelineConnector from '@mui/lab/TimelineConnector';
 import TimelineContent from '@mui/lab/TimelineContent';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { SuggestChangesetState } from '../suggestChanges.types';
+interface ISuggestChangeTimelineProps {
+    state: SuggestChangesetState;
+}
 
-export const SuggestedChangeTimeline: FC = () => {
+interface ITimelineData {
+    title: string;
+    active: boolean;
+}
+
+const StyledPaper = styled(Paper)(({ theme }) => ({
+    marginTop: theme.spacing(2),
+    borderRadius: `${theme.shape.borderRadiusLarge}px`,
+}));
+
+const StyledBox = styled(Box)(({ theme }) => ({
+    padding: theme.spacing(2),
+    marginBottom: `-${theme.spacing(4)}`,
+}));
+
+const StyledTimeline = styled(Timeline)(() => ({
+    [`& .${timelineItemClasses.root}:before`]: {
+        flex: 0,
+        padding: 0,
+    },
+}));
+
+export const SuggestedChangeTimeline: FC<ISuggestChangeTimelineProps> = ({
+    state,
+}) => {
+    const createTimeLineData = (
+        state: SuggestChangesetState
+    ): ITimelineData[] => {
+        const steps: SuggestChangesetState[] = [
+            'Draft',
+            'In review',
+            'Approved',
+            'Applied',
+        ];
+
+        return steps.map(step => ({
+            title: step,
+            active: step === state,
+        }));
+    };
+
+    const renderTimeline = () => {
+        const data = createTimeLineData(state);
+        const activeIndex = data.findIndex(item => item.active);
+        const hasActiveStep = activeIndex !== -1;
+
+        if (state === 'Cancelled') {
+            return createCancelledTimeline(data);
+        }
+
+        return createTimeline(data, hasActiveStep, activeIndex);
+    };
+
+    return (
+        <StyledPaper elevation={0}>
+            <StyledBox>
+                <StyledTimeline>{renderTimeline()}</StyledTimeline>
+            </StyledBox>
+        </StyledPaper>
+    );
+};
+
+const createTimeline = (
+    data: ITimelineData[],
+    hasActiveStep: boolean,
+    activeIndex: number
+) => {
+    return data.map(({ title }, index) => {
+        const notLastItem = !(index === data.length - 1);
+
+        const connector = (
+            <ConditionallyRender
+                condition={notLastItem}
+                show={<TimelineConnector />}
+            />
+        );
+
+        if (hasActiveStep && activeIndex >= index) {
+            return createTimelineItem('success', title, connector);
+        }
+
+        if (activeIndex + 1 === index) {
+            return createTimelineItem('primary', title, connector, {
+                variant: 'outlined',
+            });
+        }
+
+        return createTimelineItem('grey', title, connector);
+    });
+};
+
+const createCancelledTimeline = (data: ITimelineData[]) => {
+    return data.map(({ title }, index) => {
+        const notLastItem = !(index === data.length - 1);
```

Comment:
I wonder if this could be slightly renamed for clarity? something like the below suggestion or `hasSubsequentItems` or something? I don't feel strongly about this, but I just had to take a second to think about why we used `notLastItem`. 

```suggestion
        const shouldConnectToNextItem = index < data.length - 1;
```

Reply:
Makes sense